deps/rabbit/src/rabbit_vhost.erl

```diff
@@ -331,6 +333,23 @@ put_vhost(Name, Description, Tags0, DefaultQueueType, Trace, Username) ->
     end,
     Result.
 
+is_over_vhost_limit(Name) ->
+    Limit = rabbit_misc:get_env(rabbit, vhost_max, infinity),
+    is_over_vhost_limit(Name, Limit).
+
+is_over_vhost_limit(_Name, infinity) ->
+    ok;
+is_over_vhost_limit(Name, Limit) when is_integer(Limit) ->
+    case length(rabbit_db_vhost:list()) >= Limit of
+        false ->
+            ok;
+        true ->
+            rabbit_misc:protocol_error(precondition_failed,
```

Comment:
We cannot use `rabbit_misc:protocol_error/3` here: it is meant to be used in the context of a connection or channel, so, to communicate errors to clients.

Reply:
Yeah, makes sense. On it.